crates/yttrium/src/chain_abstraction/api/status.rs

```diff
@@ -0,0 +1,53 @@
+use relay_rpc::domain::ProjectId;
+use serde::{Deserialize, Serialize};
+
+pub const STATUS_ENDPOINT_PATH: &str = "/v1/ca/orchestrator/status";
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct StatusQueryParams {
+    pub project_id: ProjectId,
+    pub orchestration_id: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StatusResponseSuccessPending {
+    created_at: usize,
+    /// Polling interval in ms for the client
+    check_in: usize,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StatusResponseSuccessCompleted {
+    created_at: usize,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StatusResponseSuccessError {
+    created_at: usize,
+    error_reason: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", tag = "status")]
+pub enum StatusResponseSuccess {
+    Pending(StatusResponseSuccessPending),
+    Completed(StatusResponseSuccessCompleted),
+    Error(StatusResponseSuccessError),
```

Comment:
What's the difference between `StatusResponseError` and `StatusResponseSuccessError`?

StatusResponseError -> request failed some http error code?
StatusResponseSuccessError -> request ok but fulfilment operation failed?

Reply:
These enums are being merged in an upcoming release. For now though, `StatusResponseError` is for like validation errors I think and `StatusResponseSuccessError` is for things like `NO_ROUTES_AVAILABLE`

Reply:
Ok got it
Once merged we'll update them in Yttrium/WK